packages/flame/lib/src/components/mixins/notifiable.dart

```diff
@@ -0,0 +1,49 @@
+import 'package:flame/components.dart';
+import 'package:flame/game.dart';
+import 'package:meta/meta.dart';
+
+/// Makes a component capable of notify listeners of changes.
+///
+/// Notifieable components will automatically notify when
+/// new instances are added or removed to the game instance.
+///
+/// To notify internal changes of a component instance, the component
+/// should call [notifyListeners].
+mixin Notifiable on Component {
```

Comment:
This should probably be named something else, notifiable means that this could be notified, but it is the other way around, this one is notifiying others. `Notifier` doesn't seem to clash with any basic Flutter or Dart things, maybe that could be the new name?